### PR TITLE
grpcproxy: make grpc keep alive related options configurable

### PR DESCRIPTION
**Pull request by mlmhl**
Currently grpc-proxy doesn't config keep alive related options, so it will use the default values provided by the underlay gprc library. If clients uses a keep alive ping interval smaller than server's default minTime, connections between server and clients will be closed and reopened frequently.

For example, kubernetes's apiserver uses a [30s keep alive ping interval](https://github.com/kubernetes/kubernetes/blob/6273449fe06574b3b192b92fe487ddfbd16837af/staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go#L46) which is a typical value used by many clients, grpc-proxy uses the default minTime 5m, provided by [grpc](https://github.com/grpc/grpc-go/blob/b3dcc681293a029bc998fc5259252b35bfef4817/internal/transport/defaults.go#L40). As 5m is much bigger than 30s, so the connections will be closed and reopened frequently due to [too_many_pings](https://github.com/grpc/grpc-go/blob/b3dcc681293a029bc998fc5259252b35bfef4817/internal/transport/http2_server.go#L728) error.

### CODE REVIEW

**Comment by mlmhl** [comment]
@jingyih @gyuho PTAL :)

File: etcdmain/grpc_proxy.go [hunk 1]
```diff
@@ -358,11 +368,26 @@ func newGRPCProxyServer(lg *zap.Logger, client *clientv3.Client) *grpc.Server {
...
 		grpc.MaxConcurrentStreams(math.MaxUint32),
-	)
+	}
+	if grpcKeepAliveMinTime > time.Duration(0) {
+		gopts = append(gopts, grpc.KeepaliveEnforcementPolicy(keepalive.EnforcementPolicy{
+			MinTime:             grpcKeepAliveMinTime,
+			PermitWithoutStream: false,
+		}))
+	}
+	if grpcKeepAliveInterval > time.Duration(0) &&
```
Comment (xiang90) [inline etcdmain/grpc_proxy.go:382]: we probably should return an error if only of the two points are set.

Comment (mlmhl) [reply etcdmain/grpc_proxy.go]: I go through the GRPC's code again and found that these two options are independent of each other, if one of them are not set, the grpc server will fill it with the default values, see [here](https://github.com/grpc/grpc-go/blob/b3dcc681293a029bc998fc5259252b35bfef4817/internal/transport/http2_server.go#L207). So I changed `if grpcKeepAliveInterval > time.Duration(0) && grpcKeepAliveTimeout > time.Duration(0)` to `if grpcKeepAliveInterval > time.Duration(0) || grpcKeepAliveTimeout > time.Duration(0)`.

By the way, I think the [main etcd code](https://github.com/etcd-io/etcd/blob/c623f798cbd1a61dbee202da22518ba607ad14e5/embed/etcd.go#L643) should also make this adjustment.

**Comment by xiang90** [comment]
LGTM. Can you add this to the change log?

**Comment by mlmhl** [comment]
Submitted a PR to update the CHANGELOG #11790 , @xiang90  PTAL :)

**Pull request merged**